lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -673,8 +673,13 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
            << "Currently do not support non-empty yield operations inside for "
               "loops. Run --scf-for-to-while before running --scf-to-calyx.";
   }
+
   auto whileOp = dyn_cast<scf::WhileOp>(yieldOp->getParentOp());
-  assert(whileOp);
+  if (!whileOp) {
```

Comment:
```suggestion
  if (whileOp == nullptr) {
```

As a friendlier way to write this (: